src/formatexpr.jl

```diff
@@ -104,15 +104,15 @@ function find_next_entry_open(s::AbstractString, si::Int)
 end
 
 function find_next_entry_close(s::AbstractString, si::Int)
-    slen = length(s)
+    slen = endof(s)
```

Comment:
Actually, `slen` isn't used here nor below, so it could even be removed. I also suspect the use of that variable in the function above is incorrect since `search` never returns an index beyond the last valid index, but that's an issue for another PR.